test/v1/models/MarriageSpec.scala

```diff
@@ -0,0 +1,203 @@
+    partnerNino = "AA000000B",
+    birthDate = DateModel("01-01-1970")
+  )
+
+
+  "Marriage .reads" when {
+
+    "provided with all optional values" should {
+
+      "return a fully populated Marriage model" in {
+
+        maxMarriageJson(true).as[Marriage] shouldBe maxMarriageModel
+      }
+    }
+
+    "provided with no optional values" should {
+
+      "return a Marriage model with only mandatory items" in {
+
+        minMarriageJson(true).as[Marriage] shouldBe minMarriageModel
+      }
+    }
+
+    "provided with invalid json" should {
+
+      "throw an JsResultException" in {
+
+        val json = Json.obj("bad" -> "json")
+
+        json.validate[Marriage].isError shouldBe true
+      }
+    }
+  }
+
+  "Marriage .writes" when {
+
+    "provided with all optional values" should {
+
+      "correctly parse to JSON" in {
+
+        Json.toJson(maxMarriageModel) shouldBe maxMarriageJson(false)
+      }
+    }
+
+    "provided with no optional values" should {
+
+      "correctly parse to JSON" in {
+
+        Json.toJson(minMarriageModel) shouldBe minMarriageJson(false)
+      }
+    }
+  }
+
+  "Marriage .maritalStatusValidation" when {
+
+    "provided with an optional int" which {
+
+      "is within the valid range" should {
+
+        "return true" in {
+
+          val result = Marriage.maritalStatusValidation(Some(6))
+
+          result shouldBe true
+        }
+      }
+
+      "is on the lower allowed limit" should {
+
+        "return true" in {
+
+          val result = Marriage.maritalStatusValidation(Some(0))
+
+          result shouldBe true
+        }
+      }
+
+      "is on the upper allowed limit" should {
+
+        "return true" in {
+
+          val result = Marriage.maritalStatusValidation(Some(12))
+
+          result shouldBe true
+        }
+      }
+
+      "is below the lower allowed limit" should {
+
+        "return false" in {
+
+          val result = Marriage.maritalStatusValidation(Some(-1))
+
+          result shouldBe false
+        }
+      }
+
+      "is above the upper allowed limit" should {
+
+        "return false" in {
+
+          val result = Marriage.maritalStatusValidation(Some(13))
+
+          result shouldBe false
+        }
+      }
+    }
+
+    "not provided with the optional int" should {
+
+      "return true" in {
+
+        val result = Marriage.maritalStatusValidation(None)
+
+        result shouldBe true
+      }
+    }
+  }
+
+  "Marriage. .stringValidation" when {
+
+    "provided with a valid string" should {
+
+      "return true" in {
+
+        val result = Marriage.stringValidation(item = Some("Name"), itemName = "example")
+
+        result shouldBe true
+      }
+    }
+
+    "provided with a string which does not match the regex" should {
+
+      "return false" in {
+
+        val result = Marriage.stringValidation(item = Some("!incorrectName!"), itemName = "example")
+
+        result shouldBe false
+      }
+    }
```

Comment:
I know we're rehashing the same old test, against the same regex, but would it be possible to get a test for the string sizes please?